frontend/src/app/connection/DefaultStreamlitEndpoints.ts

```diff
@@ -165,8 +194,19 @@ export class DefaultStreamlitEndpoints implements StreamlitEndpoints {
     url: string,
     params: AxiosRequestConfig
   ): Promise<R> {
-    const serverURI = this.requireServerUri()
-    params.url = buildHttpUri(serverURI, url)
+    // If the URL starts with a scheme, assume it's an absolute URL and don't
+    // change it. Otherwise, it's a relative URL that needs to be made
+    // absolute.
+    //
+    // TODO(vdonato): Think about whether this is a reasonable long-term
+    // solution before this feature branch is merged into `develop`. If it is,
+    // then add some unit tests for this.
+    if (/^[a-z0-9]+:\/\//.test(url)) {
```

Comment:
Ah, good point. There's no reason to have `csrfRequest` do this anymore since only file-uploader related methods use  this function anyway. Removed this `if`/`else` code block.